lib/interpreters/periodic_visits.rb

```diff
@@ -456,67 +450,69 @@ def compute_possible_days(vrp)
       }
     end
 
-    def save_relations(vrp, relation_type)
-      vrp.relations.select{ |r| r.type == relation_type }.collect{ |r|
-        {
-          type: r.type,
-          linked_vehicle_ids: r.linked_vehicle_ids,
-          lapse: r.lapse,
-          periodicity: r.periodicity
-        }
+    def save_vehicle_linking_relations(vrp)
+      vrp.relations.select{ |r|
+        [:vehicle_group_duration, :vehicle_group_duration_on_weeks, :vehicle_group_duration_on_months,
+         :vehicle_trips].include?(r.type)
       }
     end
 
-    def get_all_vehicles_in_relation(relations)
-      relations&.each{ |r|
-        next if r[:type] == :vehicle_group_duration
+    def cut_linking_vehicle_relation_by_period(relation, periods, relation_type)
+      additional_relations = []
+      vehicles_in_relation =
+        relation[:linked_vehicle_ids].flat_map{ |v| @equivalent_vehicles[v] }
 
-        new_list = []
-        r[:linked_vehicle_ids].each{ |v|
-            new_list.concat(@equivalent_vehicles[v])
-        }
-        r[:linked_vehicle_ids] = new_list
-      }
+      while periods.any?
+        days_in_period = periods.slice!(0, relation.periodicity).flatten
+        relation_vehicles = vehicles_in_relation.select{ |id| days_in_period.include?(id.split('_').last.to_i) }
+        next unless relation_vehicles.any?
+
+        additional_relations << Models::Relation.new(
+          linked_vehicle_ids: relation_vehicles,
+          lapse: relation.lapse,
+          type: relation_type
+        )
+      end
+
+      additional_relations
+    end
+
+    def collect_weeks_in_schedule
+      all_days = (@schedule_start..@schedule_end).to_a
+      current_index = 0
+      next_monday = all_days.find_index{ |d| d % 7 == 0 && d > current_index }
```

Comment:
Its seems to me that the find loop is unnecessary.

We only have to care about the first and the last week. The weeks between are all complete

Reply:
yes, maybe I can simplify this function